scripts/cat/status.py

```diff
@@ -0,0 +1,639 @@
+
+        # group assignment via social
+        # we assume a clancat is the player's as default
+        # otherwise if the cat isn't a clancat, then we assume no group
+        if social == CatSocial.CLANCAT and not group:
+            new_history["group"] = CatGroup.PLAYER_CLAN
+
+        # next, we double-check that the rank is appropriate for the social, this is mostly for loner/rogue/kittypet
+        if social != self.social_lookup[rank]:
+            # getting ranks according to social category
+            possible_ranks = [
+                rank
+                for rank in self.social_lookup.keys()
+                if self.social_lookup.get(rank) == social
+            ]
+
+            new_history["rank"] = choice(possible_ranks)
+
+        self.group_history = [new_history]
+
+    def _start_standing(self):
+        """
+        Generates basic standing info for a cat. If the cat is part of a group, it creates a MEMBER dict, else it
+        creates a KNOWN standing dict for the player's clan.
+        """
+        if self.group:
+            self.standing_history = [
+                {"group": self.group, "standing": [CatStanding.MEMBER], "near": True}
+            ]
+        else:
+            self.standing_history = [
+                {
+                    "group": CatGroup.PLAYER_CLAN,
+                    "standing": [CatStanding.KNOWN],
+                    "near": True,
+                }
+            ]
+
+    # PROPERTIES
+    @property
+    def social(self) -> CatSocial:
+        """
+        Returns the cat's current social category, aka what the cat is considered by other cats within the world
+        """
+        return self.all_socials[-1]
+
+    @property
+    def all_socials(self) -> list:
+        """
+        Returns a list of all social classes the cat has been part of or is currently part of.
+        """
+        social_history_dupes = [
+            self.social_lookup[record["rank"]] for record in self.group_history
+        ]
+        social_groups = [k for k, g in groupby(social_history_dupes)]
+
+        return social_groups
+
+    @property
+    def group(self) -> CatGroup:
+        """
+        Returns the group that a cat is currently affiliated with.
+        """
+        return self.group_history[-1]["group"]
+
+    @property
+    def all_groups(self) -> list:
+        """
+        Returns a list of all groups the cat has been a part of or is currently a part of.
+        """
+        groups = []
+        for record in self.group_history:
+            if record["group"] not in groups:
+                groups.append(record["group"])
+
+        return [record["group"] for record in self.group_history]
+
+    @property
+    def rank(self) -> CatRank:
+        """
+        Returns the rank that a cat currently holds within their group.
+        """
+        rank = [
+            rank for rank in list(CatRank) if rank == self.group_history[-1]["rank"]
+        ]
+        return rank[0]
```

Comment:
```suggestion
        return CatRank(self.group_history[-1]["rank"])
```
I believe should work? I can't find documentation on what error this throws if it's not a valid Enum, but in fairness it probably should crash if it's invalid in this case anyway